### PR TITLE
topic/dynamicWrite : Changes to support dynamic write operation in es…

### PR DESCRIPTION
**Pull request by shlok7296**
… connector

### CODE REVIEW

File: src/main/java/io/confluent/connect/elasticsearch/DataConverter.java [hunk 1]
```diff
@@ -16,22 +16,19 @@
...
-import org.apache.kafka.connect.data.ConnectSchema;
-import org.apache.kafka.connect.data.Date;
-import org.apache.kafka.connect.data.Decimal;
-import org.apache.kafka.connect.data.Field;
-import org.apache.kafka.connect.data.Schema;
-import org.apache.kafka.connect.data.SchemaBuilder;
-import org.apache.kafka.connect.data.Struct;
-import org.apache.kafka.connect.data.Time;
-import org.apache.kafka.connect.data.Timestamp;
+import org.apache.kafka.connect.data.*;
```
Comment (shikhasomani) [inline src/main/java/io/confluent/connect/elasticsearch/DataConverter.java:19]: Remove this import and add individual classes to import

File: src/main/java/io/confluent/connect/elasticsearch/DataConverter.java [hunk 2]
```diff
@@ -16,22 +16,19 @@
...
 import org.apache.kafka.connect.errors.ConnectException;
 import org.apache.kafka.connect.errors.DataException;
+import org.apache.kafka.connect.header.Header;
 import org.apache.kafka.connect.json.JsonConverter;
 import org.apache.kafka.connect.sink.SinkRecord;
 import org.apache.kafka.connect.storage.Converter;
+import org.apache.kafka.connect.storage.HeaderConverter;
+import org.apache.kafka.connect.storage.SimpleHeaderConverter;
```
Comment (shikhasomani) [inline src/main/java/io/confluent/connect/elasticsearch/DataConverter.java:27]: Unused import classes

File: src/main/java/io/confluent/connect/elasticsearch/DataConverter.java [hunk 3]
```diff
@@ -184,7 +181,22 @@ public IndexableRecord convertRecord(
 
     final String payload = getPayload(record, ignoreSchema);
     final Long version = ignoreKey ? null : record.kafkaOffset();
-    return new IndexableRecord(new Key(index, type, id), payload, version);
+    Header header = record.headers().lastWithName("ss-header");
+    /*System.out.println("HEADER--------" + header);
+    SimpleHeaderConverter headerConverter = new SimpleHeaderConverter();
+    //byte[] serialized2 = headerConverter.fromConnectHeader(record.topic(), header, Schema.STRING_SCHEMA, header.value());
+    SchemaAndValue result2 = headerConverter.toConnectHeader(record.topic(), "ss-header", (byte[]) header.value());
```
Comment (shikhasomani) [inline src/main/java/io/confluent/connect/elasticsearch/DataConverter.java:188]: Require code cleanup

File: src/main/java/io/confluent/connect/elasticsearch/DataConverter.java [hunk 4]
```diff
@@ -184,7 +181,22 @@ public IndexableRecord convertRecord(
...
+    System.out.println("HEADER CONVERTED STRING-------" + result2.value());
+    //JSON_CONVERTER.fromConnectData(record.topic(), )
+    SchemaAndValue result3 = JSON_CONVERTER.toConnectData(record.topic(), (byte[]) header.value());
+    System.out.println("JSON CONVERTED VALUE ---------"+ result3.value());*/
+    String operation = null;
+    if (header != null) {
+     operation =  ((Map) header.value()).get("operation").toString();
+     System.out.println("OPERATION VALUE ---------------" + operation);
```
Comment (shikhasomani) [inline src/main/java/io/confluent/connect/elasticsearch/DataConverter.java:196]: require refactoring and handling if operation is not provided

File: src/main/java/io/confluent/connect/elasticsearch/jest/JestElasticsearchClient.java [hunk 5]
```diff
@@ -469,6 +469,10 @@ public BulkRequest createBulkRequest(List<IndexableRecord> batch) {
     if (record.payload == null) {
       return toDeleteRequest(record);
     }
+    String operation = record.operation;
+    if (operation != null) {
+      writeMethod = WriteMethod.forValue(operation);
```
Comment (shikhasomani) [inline src/main/java/io/confluent/connect/elasticsearch/jest/JestElasticsearchClient.java:474]: Make it case insensitive

File: src/main/java/io/confluent/connect/elasticsearch/DataConverter.java (hunk 1 above)
Comment (shlok7296) [reply src/main/java/io/confluent/connect/elasticsearch/DataConverter.java]: had resolved these changes in my forked repo but seems like the commits were missing while merging, have resolved them in this PR.

**Comment by ashish437** [comment]
Raised another PR for keeping confluent version in sync : https://github.com/SSI-Avalon/Kafka-Connect-Elasticsearch/pull/3

**Pull request closed without merging**